diff --git a/Documentation/Makefile b/Documentation/Makefile
index ddd3657..b018681 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -40,7 +40,7 @@ $(SRC): $(FOOTER)
 %.8: %.txt
 	$(call msg-gen, $@)
 	$(Q) $(ASCIIDOC) -b docbook -d manpage -o $(patsubst %.8,%.xml,$@) $<
-	$(Q) $(XMLTO) man --skip-validation $(patsubst %.8,%.xml,$@) 2>/dev/null
+	$(Q) $(XMLTO) --skip-validation man $(patsubst %.8,%.xml,$@) 2>/dev/null
 
 %.ps: %.8
 	$(call msg-gen, $@)
diff --git a/Makefile b/Makefile
index 037d38f..bea360f 100644
--- a/Makefile
+++ b/Makefile
@@ -177,21 +177,6 @@ test: zdtm
 	$(Q) MAKEFLAGS= $(MAKE) -C test
 .PHONY: test
 
-#
-# Generating tar requires tag matched CRIU_VERSION.
-# If not found then simply use GIT's describe with
-# "v" prefix stripped.
-head-name := $(shell git tag -l v$(CRIU_VERSION))
-ifeq ($(head-name),)
-        head-name := $(shell git describe)
-endif
-tar-name := $(shell echo $(head-name) | sed -e 's/^v//g')
-criu-$(tar-name).tar.bz2:
-	git archive --format tar --prefix 'criu-$(tar-name)/' $(head-name) | bzip2 > $@
-dist tar: criu-$(tar-name).tar.bz2
-	@true
-.PHONY: dist tar
-
 tags:
 	$(call msg-gen, $@)
 	$(Q) $(RM) tags
diff --git a/criu/Makefile.version b/criu/Makefile.version
index 9b5a614..c2194ab 100644
--- a/criu/Makefile.version
+++ b/criu/Makefile.version
@@ -1,21 +1,6 @@
 include $(__nmk_dir)msg.mk
 
 VERSION_HEADER		:= include/version.h
-GITID_FILE		:= ../.gitid
-GITID			:= $(shell if [ -d "../.git" ]; then cd .. && git describe; fi)
-
-ifeq ($(GITID),)
-        GITID := 0
-else
-        GITID_FILE_VALUE := $(shell if [ -f '$(GITID_FILE)' ]; then if [ `cat '$(GITID_FILE)'` = $(GITID) ]; then echo y; fi; fi)
-        ifneq ($(GITID_FILE_VALUE),y)
-                .PHONY: $(GITID_FILE)
-        endif
-endif
-
-$(GITID_FILE):
-	$(call msg-gen, $@)
-	$(Q) echo "$(GITID)" > $(GITID_FILE)
 
 $(VERSION_HEADER): ../Makefile.versions $(GITID_FILE)
 	$(call msg-gen, $@)
